subsys/bluetooth/controller/hci/hci.c

```diff
@@ -4876,18 +4897,54 @@ static void dup_ext_adv_mode_reset(struct dup_ext_adv_mode *dup_adv_mode)
 }
 
 #if defined(CONFIG_BT_CTLR_SYNC_PERIODIC_ADI_SUPPORT)
-static void dup_ext_adv_reset(int count)
+static void dup_ext_adv_reset(void)
 {
 	int i;
 
-	for (i = 0; i < count; i++) {
+	for (i = 0; i < dup_count; i++) {
 		struct dup_entry *dup;
 
 		dup = &dup_filter[i];
 		dup->mask = 0U;
 		dup_ext_adv_mode_reset(dup->adv_mode);
 	}
 }
+
+static void dup_periodic_adv_reset(uint8_t addr_type, const uint8_t *addr,
+				   uint8_t sid)
+{
+	int addr_idx;
+
+	for (addr_idx = 0; addr_idx < dup_count; addr_idx++) {
+		struct dup_ext_adv_mode *dup_mode;
+		struct dup_entry *dup;
+		int set_idx;
```

Comment:
```suggestion
		uint16_t set_idx;
```
If you want `set_idx` to be same type as `dup_mode->set_count` which it is compared against.